inspire_json_merger/postprocess.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2020 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+from __future__ import absolute_import, division, print_function
+
+import itertools
+import json
+from operator import itemgetter
+
+from json_merger.conflict import Conflict
+
+from inspire_json_merger.utils import ORDER_KEY
+
+
+def postprocess_results(merged, conflicts):
+    """Run all postprocessing to provide output understandable by record-editor.
+    Args:
+        merged(dict): Merged document
+        conflicts(list): List of all possible conflicts
+
+    Returns: A tuple containing the resulted merged record in json format and a
+        an list containing all generated conflicts.
+
+    """
+    conflicts, merged = postprocess_conflicts(conflicts, merged)
+    conflicts_as_json = [json.loads(c.to_json()) for c in conflicts]
+    flat_conflicts_as_json = remove_ordering_from_authors_conflicts(
+        list(itertools.chain.from_iterable(conflicts_as_json)))
+    merged = remove_ordering_from_authors_merged(merged)
+
+    return merged, flat_conflicts_as_json
+
+
+def remove_ordering_from_authors_conflicts(conflicts):
+    """Sorts authors and cleans up ordering information from conflicts."""
+    new_conflicts = []
+    for conflict in conflicts:
+        if '/authors' in conflict["path"]:
+            del conflict["value"][ORDER_KEY]
+        new_conflicts.append(conflict)
+    return new_conflicts
+
+
+def remove_ordering_from_authors_merged(merged):
+    """Sorts authors and cleans up ordering information from merged record."""
+    if 'authors' in merged:
+        merged["authors"].sort(key=itemgetter(ORDER_KEY))
```

Comment:
why are you doing this? the ordering is determined by the merger and is significant, you shouldn't be sorting here.

Reply:
I'm sorting here by ORDER_KEY as when we insert authors from HEAD they will be put at the end of the list no mater what order position is determined by ORDER_KEY they have.

Reply:
Still is should be the same as order determined by merger only this new entries from HEAD will be sorted in correct positions

Reply:
Discussed IRL